### ISSUE TITLE
Shape `box.shapeDecoration.shape.beveledRectangle()` isn't working

### ISSUE THREAD

**Issue by Azzeccagarbugli** [conceptadev/mix#256]
### Package version

**`1.0.0-beta.12`**

### Flutter version

```
Flutter 3.19.6 • channel stable • https://github.com/flutter/flutter.git
Framework • revision 54e66469a9 (3 weeks ago) • 2024-04-17 13:08:03 -0700
Engine • revision c4cd48e186
Tools • Dart 3.3.4 • DevTools 2.31.1
```

### Steps to reproduce

1. Call `box.shapeDecoration.shape.beveledRectangle()`

### Expected results

I would expect the code to work and properly apply the `beveledRectangle` shape.

### Actual results

`_AssertionError ('package:mix/src/attributes/decoration/decoration_dto.dart': Failed assertion: line 68 pos 12: 'isMergeable()': This decoration is not mergeable with other)`

### Code sample

<details open><summary>Code sample</summary>

```dart
CustomButtonSize.large(
        box.shapeDecoration.shape.beveledRectangle(),
        box.height(60),
        box.width(350),
        box.minWidth(350),
        box.color($colors.primary.resolve(context)),
        box.padding.all($space.medium.resolve(context)),
      ),
```

</details>

**Comment by leoafarias** [comment]
@Azzeccagarbugli We are releasing the latest version of the beta later today. This happens because `box.color` uses a BoxDecoration and ShapeDecoration, so they conflict with each other. I suggest using `box.shapeDecoration.color`.

**Comment by Azzeccagarbugli** [comment]
> @Azzeccagarbugli We are releasing the latest version of the beta later today. This happens because `box.color` uses a BoxDecoration and ShapeDecoration, so they conflict with each other. I suggest using `box.shapeDecoration.color`.


Uh ok, that makes a lot of sense, but I still get the same error even if I do the following:

```dart
CustomButtonSize.large(
  box.height(60),
  box.width(350),
  box.minWidth(350),
  box.padding.all($space.medium.resolve(context)),
  box.shapeDecoration.shape.beveledRectangle(),
),
CustomButtonType.primary(
  box.shapeDecoration.color($colors.primary.resolve(context)),
),
```

Now I'm just applying the color of the `box` by using the color from the `shapeDecoration` so it should be fine.

Am I missing something? 🤔

**Comment by Azzeccagarbugli** [comment]
Sure, thanks a lot once again for the rapid support and help 🫶🏻

**Comment by leoafarias** [comment]
Is addressed by #264
